apps/examples/hello/hello_main.c

```diff
@@ -68,5 +70,46 @@ int hello_main(int argc, char *argv[])
 #endif
 {
 	printf("Hello, World!!\n");
+	/* Call stay IOCTL to keep PM domain in current state because maybe we know we need to do some
+	 * non-frequent work with time gaps and dont want the board to sleep in between
+	 */
+	pm_ioctl(PM_IOC_STAY);
+
+	{
+		//The work happens
+	}
+
+	/* The work is done, now relax the stay on the current state of the domain */
+	pm_ioctl(PM_IOC_RELAX);
+
+	{
+		//App does some other non-blocking work
+	}
+
+	/* Now, maybe the app has completed the major part of its work and is suggesting that the board
```

Comment:
Yes, I agree with your point here, we should be able to check the stay count to determine the status of the current state.

Reply:
Here, the app can check the staycount and if found = 0, it can try to invoke sleep.
OR
App can just try to invoke sleep and if the staycount is 0, sleep request may be granted or else denied (This will happen in PM module itself, app does not have to worry about staycounts).

Reply:
Ya, I agree with the second statement. App does not need to take care of any PM related variable or behavior changes, it should only take care of it's status, to decide whether to keep the state or relax the state. The permission for state changes should be handled by PM logic.